model/extensions/Zicsr/zicsr_insts.sail

```diff
@@ -29,9 +29,12 @@ mapping clause encdec = CSRImm(csr, imm, rd, op)
   when currentlyEnabled(Ext_Zicsr)
 
 function doCSR(csr : csreg, rs1_val : xlenbits, rd : regidx, op : csrop, is_CSR_Write: bool) -> ExecutionResult = {
-  if not(check_CSR(csr, cur_privilege, is_CSR_Write))
-  then Illegal_Instruction()
-  else if not(ext_check_CSR(csr, cur_privilege, is_CSR_Write))
+  match check_CSR(csr, cur_privilege, is_CSR_Write) {
+    CSRAccessSuccess => (),
+    CSRAccessIllegal => return Illegal_Instruction(),
+    CSRNotExist => return Illegal_Instruction(),
```

Comment:
So currently these have exactly the same effect? What will the difference be in future? I think we need a comment for that.